code/ai/aigoals.h

```diff
@@ -103,14 +105,14 @@ extern int ai_goal_num(ai_goal *goals);
 extern void ai_add_ship_goal_scripting(int mode, int submode, int priority, char *shipname, ai_info *aip);
 extern void ai_add_ship_goal_sexp( int sexp, int type, ai_info *aip );
 extern void ai_add_wing_goal_sexp( int sexp, int type, wing *wingp );
-extern void ai_add_goal_sub_sexp( int sexp, int type, ai_goal *aigp, char *actor_name );
+extern void ai_add_goal_sub_sexp( int sexp, int type, ai_goal *aigp, char *actor_name);
 
 extern int ai_remove_goal_sexp_sub( int sexp, ai_goal* aigp );
 extern void ai_remove_wing_goal_sexp( int sexp, wing *wingp );
 
 // adds goals to ships/sings through player orders
-extern void ai_add_ship_goal_player( int type, int mode, int submode, char *shipname, ai_info *aip );
-extern void ai_add_wing_goal_player( int type, int mode, int submode, char *shipname, int wingnum );
+extern void ai_add_ship_goal_player( int type, int mode, int submode, char *shipname, ai_info *aip, object_ship_wing_point_team lua_target = object_ship_wing_point_team());
+extern void ai_add_wing_goal_player( int type, int mode, int submode, char *shipname, int wingnum, object_ship_wing_point_team lua_target = object_ship_wing_point_team());
```

Comment:
I'm not really a fan of copying an `object_ship_wing_point_team` struct everywhere, since it will only be used if in fact there is a Lua AI target.  That said I don't want to rule it out either.  What are the tradeoffs between doing it this way and, say, storing an index into a global vector of OSWPTs?

Reply:
Indeed, that's bad. Changed to const references